content/solids/components/voxel-painter.ts

```diff
@@ -0,0 +1,462 @@
+        especially for save button
+        Talk about how to deal with unclicking mouse when not on canvas
+      eraser texture
+      Want to click once and have just one thing appear
+      Try orthographic
+      More color variation with sides?
+
+      Pull from master
+      Create pull request
+     */
+
+    await loadScript('/resources/shared/vendor/three-91.min.js');
+    
+    const TAU = Math.PI * 2;
+    const HS3 = Math.sqrt(3) / 2;
+    const zUnit = new THREE.Vector3(0, 0, 1);
+    const yUnit = new THREE.Vector3(0, 1, 0);
+    const xUnit = new THREE.Vector3(1, 0, 0);
+
+    const zeroVector = new THREE.Vector3(0., 0., 0.);
+    const zeroMatrix = new THREE.Matrix4().set(0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.);
+
+    const clock = new THREE.Clock(true);
+    var frameCount = 0
+    var frameDelta = 1. / 60.
+
+    const v1 = new THREE.Vector3();
+    const v2 = new THREE.Vector3();
+    const v3 = new THREE.Vector3();
+    const m1 = new THREE.Matrix4();
+    const q1 = new THREE.Quaternion();
+    const pl = new THREE.Plane();
+
+    const log = console.log;
+    const scene = new THREE.Scene();
+
+    let rotateOnly = this.attr('rotateOnly')
+    if (rotateOnly === undefined)
+      rotateOnly = false
+
+    const renderer = new THREE.WebGLRenderer({antialias: true});
+    renderer.localClippingEnabled = true;
+    renderer.setClearColor(0xffffff);
+    scene.background = new THREE.Color(0xf0f0f0);
+    renderer.setSize(400, 400);
+
+    const updateFunctions: (() => void)[] = [];
+    const objectsOnWhichVoxelsCanBePlaced: THREE.Object3D[] = [];
+
+    const $canvas = $(renderer.domElement) as CanvasView;
+    this.append($canvas);
+
+    //Save button, TODO not appearingS
+    if (this.attr('showSaveButton') === "true")
+    {
+      let $button = $(document.createElement("BUTTON")) as ElementView
+      log($button)
+      this.append($button)
+
+      $button.on('click', () => {
+        let outputString = '"'
+        for (let i = 0; i < voxels.length; i++)
+          for (let j = 0; j < 3; j++)
+            outputString += voxels[i].position.getComponent(j) + ","
+        outputString += '"'
+        prompt('In order to load the current voxels in, copy the below, then paste it next to shape= ', outputString)
+      })
+    }
+
+    {
+      const ambientLight = new THREE.AmbientLight(0x606060);
+      scene.add(ambientLight);
+
+      const directionalLight = new THREE.DirectionalLight(0xffffff);
+      directionalLight.position.set(1, 0.75, 0.5).normalize();
+      scene.add(directionalLight);
+    }
+
+    let rendererSize = renderer.getSize()
+    // const camera = new THREE.PerspectiveCamera(45, rendererSize.width / rendererSize.height, 1., 10000.);
+    const cameraW = 30.
+    const cameraH = cameraW * rendererSize.height /rendererSize.width
+    const camera = new THREE.OrthographicCamera(-cameraW / 2., cameraW / 2., cameraH / 2., -cameraH / 2., 1., 10000.)
```

Comment:
Can we have an attribute to switch between the different cameras, e.g. `const camera = this.hasAttr('othographic') ? new THREE.OrthographicCamera() : new THREE.PerspectiveCamera ()`?

Reply:
Done

Reply:
Done

Reply:
Done